lib/qbwc/request/discountproducts.rb

```diff
@@ -33,17 +33,29 @@ def generate_request_queries(objects, params)
             config = { connection_id: params['connection_id'] }.with_indifferent_access
             session_id = Persistence::Session.save(config, object)
 
-            request << search_xml(product_identifier(object), session_id)
+            if object['list_id'].to_s.empty?
+              request << search_xml_by_name(product_identifier(object), session_id)
+            else
+              request << search_xml_by_id(object['list_id'], session_id)
+            end
           end
         end
 
-        def search_xml(product_id, session_id)
+        def search_xml_by_id(object_id, session_id)
+          <<~XML
+            <ItemDiscountQueryRq requestID="#{session_id}">
+              <ListID>#{object_id}</ListID>
+            </ItemDiscountQueryRq>
+          XML
+        end
+
+        def search_xml_by_name(object_id, session_id)
           <<~XML
             <ItemDiscountQueryRq requestID="#{session_id}">
-              <MaxReturned>10000</MaxReturned>
+              <MaxReturned>50</MaxReturned>
```

Comment:
Why was this one changed to 50 while the others stayed at 10000?

Reply:
Oh - I should keep them the same. Not sure why. Good eye